### PR TITLE
Migrate orchestrator config to its own struct/package

### PR DESCRIPTION
**Pull request by mfpierre**
### What does this PR do?

Migrate orchestrator config to its own struct/package

### Motivation

What inspired you to submit this pull request?

### Additional Notes

Anything else we should know when reviewing?

### Describe your test plan

Write there any instructions and details you may have to test your PR.

### CODE REVIEW

**Review by nammn — COMMENTED** [review]
💯 lgtm, just a few nits and things I would like to understand 👍

File: pkg/process/config/config.go [hunk 1]
```diff
@@ -371,7 +366,8 @@ func NewAgentConfig(loggerName config.LoggerName, yamlPath, netYamlPath string)
 	}
 
 	// activate the pod collection if enabled and we have the cluster name set
-	if cfg.OrchestrationCollectionEnabled && cfg.KubeClusterName != "" {
+	// todo use orchestration config
```
[the remark this answers is not in the document]
Comment (nammn) [reply pkg/process/config/config.go:369]: nvm. This comment can be removed, no ?

File: pkg/orchestrator/config/config_test.go [hunk 2]
```diff
@@ -1,11 +1,17 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-2020 Datadog, Inc.
+
 package config
 
 import (
+	"net/url"
```
[the remark this answers is not in the document]
Comment (mfpierre) [reply pkg/orchestrator/config/config_test.go:9]: yes it is part of the process check interface that only passes the process config to the check https://github.com/DataDog/datadog-agent/blob/1ee0bc9aceba46b0968b1412e771ae7e0890169e/pkg/process/checks/checks.go#L16

File: pkg/process/config/config.go [hunk 3]
```diff
@@ -227,6 +216,9 @@ func NewDefaultAgentConfig(canAccessContainers bool) *AgentConfig {
 		EnableTracepoints:            false,
 		CollectDNSStats:              true,
 
+		// Orchestrator config
+		Orchestrator: oconfig.NewDefaultOrchestratorConfig(),
```
[the remark this answers is not in the document]
Comment (mfpierre) [reply pkg/process/config/config.go:220]: yes, it is part of the process check interface that only passes the process config to the check https://github.com/DataDog/datadog-agent/blob/1ee0bc9aceba46b0968b1412e771ae7e0890169e/pkg/process/checks/checks.go#L16

File: pkg/orchestrator/redact/data_scrubber_test.go [hunk 4]
```diff
@@ -148,39 +144,22 @@ func benchmarkCommandMatching(nbCommands int, b *testing.B) {
...
-	b.Run(fmt.Sprintf("simplified"), func(b *testing.B) {
+	b.Run("simplified", func(b *testing.B) {
 		for n := 0; n < b.N; n++ {
 			for _, p := range runningProcesses {
 				_, c = scrubber.ScrubSimpleCommand(p)
 			}
 		}
 	})
 
-	b.Run(fmt.Sprintf("default"), func(b *testing.B) {
```
[the remark this answers is not in the document]
Comment (mfpierre) [reply pkg/orchestrator/redact/data_scrubber_test.go:176]: the issue is that's it's creating an import cycle 🙈 not sure if it's worth keeping the benchmark in this state

```
package github.com/DataDog/datadog-agent/pkg/orchestrator/redact (test)
	imports github.com/DataDog/datadog-agent/pkg/process/config
	imports github.com/DataDog/datadog-agent/pkg/orchestrator/config
	imports github.com/DataDog/datadog-agent/pkg/orchestrator/redact
```

[the remark this answers is not in the document]
Comment (nammn) [reply pkg/orchestrator/redact/data_scrubber_test.go:176]: Ahh, then it is not worth. We can remove it 👍

**Review by robertjli — APPROVED** [review]

File: pkg/clusteragent/orchestrator/controller.go [hunk 5]
```diff
@@ -151,10 +140,8 @@ func newController(ctx ControllerContext) (*Controller, error) {
 		clusterName:             ctx.ClusterName,
 		clusterID:               clusterID,
 		orchestratorConfig:      orchestratorCfg,
-		simpleDataScrubber:      orchestrator.NewDefaultDataScrubber(),
 		forwarder:               forwarder.NewDefaultForwarder(podForwarderOpts),
 		isLeaderFunc:            ctx.IsLeaderFunc,
-		isScrubbingEnabled:      config.Datadog.GetBool("orchestrator_explorer.container_scrubbing.enabled"),
 		extraTags:               config.Datadog.GetStringSlice("orchestrator_explorer.extra_tags"),
```
Comment (nammn) [inline pkg/clusteragent/orchestrator/controller.go:145]: do we want to move the extraTags also under `orchestratorCfg`?
Would be one less parameter in the other functions 🤔

File: pkg/orchestrator/pod.go [hunk 6]
```diff
@@ -28,13 +28,12 @@ import (
...
 
 const (
-	redactedValue = "********"
 	// from https://github.com/kubernetes/kubernetes/blob/abe6321296123aaba8e83978f7d17951ab1b64fd/pkg/util/node/node.go#L43
 	nodeUnreachablePodReason = "NodeLost"
 )
 
 // ProcessPodList processes a pod list into process messages
-func ProcessPodList(podList []*v1.Pod, groupID int32, hostName string, clusterName string, clusterID string, withScrubbing bool, maxPerMessage int, scrubber *DataScrubber, extraTags []string) ([]model.MessageBody, error) {
+func ProcessPodList(podList []*v1.Pod, groupID int32, hostName string, clusterID string, config *config.OrchestratorConfig, extraTags []string) ([]model.MessageBody, error) {
```
Comment (nammn) [inline pkg/orchestrator/pod.go:36]: nit: is there a reason you call this here config and in the other processer (DCA) places cfg? Is that on purpose?

**Review by nammn — APPROVED** [review]
lgtm! 1 small thing and one nit.

**Pull request merged**